main.tf

```diff
@@ -195,3 +211,34 @@ resource "aws_eks_node_group" "default" {
     ignore_changes = [scaling_config[0].desired_size]
   }
 }
+
+resource "aws_security_group" "remote_access" {
+  count       = local.enabled && var.ec2_ssh_key != null ? 1 : 0
+  name        = "eks-${var.cluster_name}-${module.label.id}-remote-access"
+  description = "Security group for all nodes in the nodeGroup to allow SSH access"
+  vpc_id      = data.aws_subnet.default.vpc_id
+  tags        = module.label.tags
+}
+
+resource "aws_security_group_rule" "public_ssh" {
+  count             = local.enabled && var.ec2_ssh_key != null && length(var.source_security_group_ids) < 1 ? 1 : 0
+  from_port         = 22
+  protocol          = "tcp"
+  security_group_id = aws_security_group.remote_access[0].id
+  to_port           = 22
+  type              = "ingress"
+
+  cidr_blocks = [
+    "0.0.0.0/0"
```

Comment:
this emulates the default behavior of non-launch template node groups that specify a `remote_access` ssh key but no source security group.